nixui/graphics/navbar.py

```diff
@@ -0,0 +1,71 @@
+from functools import partial
+
+from PyQt5 import QtWidgets
+
+from nixui.options.attribute import Attribute
+
+
+class FocusChangeTextLineEdit(QtWidgets.QLineEdit):
+    def __init__(self, unfocused_text, focused_text, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.unfocused_text = unfocused_text
+        self.focused_text = focused_text
+        self.setText(unfocused_text)
+
+    def focusInEvent(self, event):
+        super().focusInEvent(event)
+        self.setText(self.focused_text)
+
+    def focusOutEvent(self, event):
+        super().focusOutEvent(event)
+        self.setText(self.unfocused_text)
+
+
+class NavBar(QtWidgets.QWidget):
+    """
+    Horizontal Layout:
+    - Back arrow (previous path)
+    - Up arrow (up a path)
+    - Path Textbox: (foo.bar.baz converted to Foo > Bar > Baz until clicked)
+    - Searchbox
+
+    TODO:
+    - implement ListOf
+    - move undo toolbar item here
+    - delete search toolbar item
+    """
+    def __init__(self, option_path, set_option_fn, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.set_option_fn = set_option_fn
+
+        # create widgets and define behavior
+        back_btn = QtWidgets.QPushButton('◀')
+        back_btn.clicked.connect(lambda: print('not implemented'))
+
+        up_btn = QtWidgets.QPushButton('▲')
```

Comment:
Not so sure if we should have a back button if we already have breadcrumbs. I think an Clicked on the breadcrumb component makes more sense.

Reply:
We don't have breadcrumbs right now unfortunately. I'll make an issue for that though. 

https://github.com/nix-gui/nix-gui/issues/88

Reply:
Additionally, the back arrow should take us to the previous view, be it "search" or "options" (tree). This isn't always in the breadcrumbs.

Reply:
Right, my bad. I meant to say the "up one level" button.

Reply:
ok, resolving in favor of #88